### PR TITLE
Update to use the new Android library build rules.

### PR DESCRIPTION
**Pull request by thomas104**
---

- [ ] `./mach build -d` does not report any errors
- [ ] `./mach test-tidy` does not report any errors
- [ ] These changes fix #__ (github issue number if applicable).

- [ ] There are tests for these changes OR
- [ ] These changes do not require tests because _____

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/14003)

### CODE REVIEW

File: components/servo/build.rs [hunk 1]
```diff
@@ -17,22 +17,46 @@ fn main() {
...
+    // Options for host are:  "linux-x86_64" "linux-x86" "darwin-x86_64" "darwin-x86"
+    // per: https://android.googlesource.com/platform/ndk/+/ics-mr0/docs/STANDALONE-TOOLCHAIN.html
 
-    // Get the standalone NDK path from NDK_STANDALONE env.
+    let host = env::var("HOST").unwrap();
+    let google_host = match host.as_ref() {
+        "i686-unknown-linux-gnu" => "linux-x86",
+        "x86_64-apple-darwin" => "darwin-x86_64",
+        "x86_64-unknown-linux-gnu" => "linux-x86_64",
+        _ => panic!("Unknown support android cross-compile host: {}", host)
```
Comment (shinglyu) [inline components/servo/build.rs]: Do you mean "Unknown android cross-compile host"?

File: components/servo/build.rs [hunk 2]
```diff
@@ -17,22 +17,46 @@ fn main() {
...
+    };
+
+    let toolchain_path = ndk_path.join("toolchains").join("arm-linux-androideabi-4.9").join("prebuilt").
+        join(google_host);
```
Comment (shinglyu) [inline components/servo/build.rs]: Consider moving the `.` from the previous line to this line: 

```
foo.join('bar')
    .join('baz');
```

File: components/servo/fake-ld.sh [hunk 3]
```diff
@@ -9,7 +9,23 @@ set -o nounset
 set -o pipefail
 
 TARGET_DIR="${OUT_DIR}/../../.."
-arm-linux-androideabi-gcc "${@}" \
-                          "${LDFLAGS-}" -lc -shared \
-                          -o "${TARGET_DIR}/libservo.so"
-touch "${TARGET_DIR}/servo"
+
+export _ANDROID_ARCH=arch-arm
+export ANDROID_SYSROOT="$ANDROID_NDK/platforms/android-18/$_ANDROID_ARCH"
```
Comment (shinglyu) [inline components/servo/fake-ld.sh]: Please use the more secure form `"${ANDROID_NDK}/platforms...."` for shell variables. This also applies to other variables in this patch. (See https://github.com/servo/servo/blob/master/docs/STYLE_GUIDE.md)

File: components/servo/fake-ld.sh [hunk 4]
```diff
@@ -9,7 +9,23 @@ set -o nounset
 set -o pipefail
 
 TARGET_DIR="${OUT_DIR}/../../.."
-arm-linux-androideabi-gcc "${@}" \
-                          "${LDFLAGS-}" -lc -shared \
-                          -o "${TARGET_DIR}/libservo.so"
-touch "${TARGET_DIR}/servo"
+
+export _ANDROID_ARCH=arch-arm
```
Comment (shinglyu) [inline components/servo/fake-ld.sh]: Consider quoting this string (and the others below).

File: python/servo/build_commands.py [hunk 5]
```diff
@@ -238,6 +240,38 @@ def build(self, target=None, release=False, dev=False, jobs=None,
...
+            elif os_type == "darwin":
+                if cpu_type in ["i386", "i486", "i686", "i768", "x86"]:
+                    host = "darwin-x86"
+                elif cpu_type in ["x86_64", "x86-64", "x64", "amd64"]:
+                    host = "darwin-x86_64"
+                else:
+                    host = "darwin-unknown"
+            else:
+                raise Exception("Android cross builds are only supported on Linux and macOS.")
+            env['PATH'] = path.join(env['ANDROID_NDK'], "toolchains", "arm-linux-androideabi-4.9",
```
Comment (shinglyu) [inline python/servo/build_commands.py]: This string formatting is quite complex, I'd prefer using `.format()` like so:
```
"{andorid_path}:{path}".format(android_path=path.join(...), path=env['PATH'])
```

But I guess it's OK to leave it as-is.

**Review by shinglyu — COMMENTED** [review]
LGTM from a syntax level, but we might need someone who knows Android toolchain to check the Android bits.

**Comment by shinglyu** [comment]
Oh there are tidy errors in Travis, please fix them as well.

**Comment by jdm** [comment]
@thomas104 Are you planning to address the feedback here?

**Comment by nox** [comment]
This probably needs a new owner.

**Comment by jdm** [comment]
@fabricedesre Is this something you could take over?

**Comment by fabricedesre** [comment]
yes, I'll update that after we land https://github.com/servo/servo/pull/15773

**Comment by nox** [comment]
@fabricedesre Are you still working on this?

**Comment by fabricedesre** [comment]
No I'm not. Sorry I should have unassigned myself.

**Comment by jdm** [comment]
Closing due to inactivity; tracking in https://github.com/servo/servo/issues/19240.

**Pull request closed without merging**